### PR TITLE
[DBForMariaDb]Init for DBForMariaDb

### PR DESCRIPTION
**Pull request by wyunchi-ms**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/MariaDB/custom/Get-AzMariaDBConnectionString.ps1 [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        [Parameter(ParameterSetName='ServerObject', ValueFromPipeline, Mandatory, HelpMessage='Identity Parameter')]
+        [Microsoft.Azure.PowerShell.Cmdlets.MariaDb.Category('Path')]
+        [Microsoft.Azure.PowerShell.Cmdlets.MariaDb.Models.Api20180601Preview.IServer]
+        # Identity Parameter
+        # To construct, see NOTES section for INPUTOBJECT properties and create a hash table.
+        ${InputObject},
+
+        [Parameter(Mandatory, HelpMessage='Connect client type')]
+        [Microsoft.Azure.PowerShell.Cmdlets.MariaDb.Category('Path')]
+        [Validateset('ADO.NET', 'JDBC', 'Node.js', 'PHP', 'Python', 'Ruby', 'WebApp')]
```
Comment (dolauli) [inline src/MariaDB/custom/Get-AzMariaDBConnectionString.ps1:50]: To use argument may be better here. See https://github.com/Azure/autorest.powershell/blob/master/docs/directives.md#Argument-Completers for details.

File: src/MariaDB/custom/Restart-AzMariaDbServer.ps1 [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        [Microsoft.Azure.PowerShell.Cmdlets.MariaDb.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials}
+    )
+    
+    process {
+        try {
+            if ($PSBoundParameters.ContainsKey('InputObject')) {
+                $Fields = $InputObject.Id.Split('/')
```
Comment (dolauli) [inline src/MariaDB/custom/Restart-AzMariaDbServer.ps1:116]: It's better we could validate the Id here.

File: src/MariaDB/readme.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+> see https://aka.ms/autorest
+
+``` yaml
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
+input-file:
+  - $(repo)/specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json
+
+title: MariaDB
+module-version: 4.0.0
```
Comment (dolauli) [inline src/MariaDB/readme.md:56]: For new RP, it should be 0.0.1

File: src/MariaDB/readme.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+directive:
+  - where:
```
Comment (dolauli) [inline src/MariaDB/readme.md:59]: Not sure why renaming is needed here, because the cmdlet is hidden finally.

**Comment by wyunchi-ms** [comment]
Hi @dolauli , I have updated the code. Please review it again.

**Pull request merged**